### PR TITLE
Update ASP.NET Core Packages to 6.0.4 And Remove Coherent Dependency

### PR DESCRIPTION
**Pull request by TanayParikh**

### CODE REVIEW

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -35,46 +35,6 @@
     <Dependency Name="Microsoft.WindowsAppSDK" Version="0.0.1">
       <Uri>https://dev.azure.com/microsoft/ProjectReunion/_git/ProjectReunionInternal</Uri>
     </Dependency>
-    <Dependency Name="Microsoft.AspNetCore.App.Runtime.win-x64" Version="6.0.3" CoherentParentDependency="Microsoft.Dotnet.Sdk.Internal">
```
Comment (TanayParikh) [inline eng/Version.Details.xml:38]: From my understanding, if we're manually managing the Microsoft.AspNetCore.* dependencies, then we don't need these entries in the Version.Details.xml

File: eng/Versions.props [hunk 2]
```diff
@@ -23,16 +23,16 @@
...
-    <MicrosoftAspNetCoreAuthorizationPackageVersion>6.0.3</MicrosoftAspNetCoreAuthorizationPackageVersion>
-    <MicrosoftAspNetCoreComponentsAnalyzersPackageVersion>6.0.3</MicrosoftAspNetCoreComponentsAnalyzersPackageVersion>
-    <MicrosoftAspNetCoreComponentsFormsPackageVersion>6.0.3</MicrosoftAspNetCoreComponentsFormsPackageVersion>
-    <MicrosoftAspNetCoreComponentsPackageVersion>6.0.3</MicrosoftAspNetCoreComponentsPackageVersion>
-    <MicrosoftAspNetCoreComponentsWebPackageVersion>6.0.3</MicrosoftAspNetCoreComponentsWebPackageVersion>
-    <MicrosoftAspNetCoreComponentsWebViewPackageVersion>6.0.3</MicrosoftAspNetCoreComponentsWebViewPackageVersion>
-    <MicrosoftAspNetCoreMetadataPackageVersion>6.0.3</MicrosoftAspNetCoreMetadataPackageVersion>
-    <MicrosoftJSInteropPackageVersion>6.0.3</MicrosoftJSInteropPackageVersion>
-    <MicrosoftWindowsDesktopAppRuntimewinx64Version>6.0.3</MicrosoftWindowsDesktopAppRuntimewinx64Version>
+    <MicrosoftAspNetCoreAppRuntimewinx64Version>6.0.4</MicrosoftAspNetCoreAppRuntimewinx64Version>
```
Comment (TanayParikh) [inline eng/Versions.props:26]: Can using 6.0.4 conflict with the SDK if it's 6.0.3xx?

Comment (rmarinho) [reply eng/Versions.props:26]: I m not sure.

**Review by rmarinho — APPROVED** [review]

File: eng/Version.Details.xml [hunk 3]
```diff
@@ -35,46 +35,6 @@
...
-      <Uri>https://dev.azure.com/dnceng/internal/_git/dotnet-aspnetcore</Uri>
-      <Sha>c911002ab43b7b989ed67090f2a48d9073d5118d</Sha>
-    </Dependency>
-    <Dependency Name="Microsoft.AspNetCore.Authorization" Version="6.0.3" CoherentParentDependency="Microsoft.Dotnet.Sdk.Internal">
-      <Uri>https://dev.azure.com/dnceng/internal/_git/dotnet-aspnetcore</Uri>
-      <Sha>c911002ab43b7b989ed67090f2a48d9073d5118d</Sha>
-    </Dependency>
```
Comment (jonathanpeppers) [inline eng/Version.Details.xml:45]: If this is removed here, what will update the `$(MicrosoftAspNetCoreAuthorizationPackageVersion)` property going forward? This is basically opting out of Maestro.

Should we instead keep using Maestro and setup subscriptions like:
```
> darc add-subscription --channel ".NET 6" --target-branch main --source-repo https://github.com/dotnet/aspnetcore --target-repo https://github.com/dotnet/maui
```
I already see 6.0.5 builds on this channel.

Maybe we should chat on Teams what you're wanting to do, thanks!

File: eng/Versions.props [hunk 4]
```diff
@@ -23,16 +23,16 @@
...
+    <MicrosoftAspNetCoreAuthorizationPackageVersion>6.0.4</MicrosoftAspNetCoreAuthorizationPackageVersion>
+    <MicrosoftAspNetCoreComponentsAnalyzersPackageVersion>6.0.4</MicrosoftAspNetCoreComponentsAnalyzersPackageVersion>
+    <MicrosoftAspNetCoreComponentsFormsPackageVersion>6.0.4</MicrosoftAspNetCoreComponentsFormsPackageVersion>
+    <MicrosoftAspNetCoreComponentsPackageVersion>6.0.4</MicrosoftAspNetCoreComponentsPackageVersion>
+    <MicrosoftAspNetCoreComponentsWebPackageVersion>6.0.4</MicrosoftAspNetCoreComponentsWebPackageVersion>
+    <MicrosoftAspNetCoreComponentsWebViewPackageVersion>6.0.4</MicrosoftAspNetCoreComponentsWebViewPackageVersion>
+    <MicrosoftAspNetCoreMetadataPackageVersion>6.0.4</MicrosoftAspNetCoreMetadataPackageVersion>
+    <MicrosoftJSInteropPackageVersion>6.0.4</MicrosoftJSInteropPackageVersion>
+    <MicrosoftWindowsDesktopAppRuntimewinx64Version>6.0.4</MicrosoftWindowsDesktopAppRuntimewinx64Version>
```
Comment (jonathanpeppers) [inline eng/Versions.props:35]: If we're going to stop using Maestro, should we collapse these into a single property? `$(AspNetCoreVersion)` or something?

File: eng/Version.Details.xml (hunk 1 above)
Comment (TanayParikh) [reply eng/Version.Details.xml:38]: Added back here. Will setup a subscription (per https://github.com/dotnet/maui/pull/6014#discussion_r849500075) once this PR is merged.

File: eng/Versions.props (hunk 2 above)
Comment (TanayParikh) [reply eng/Versions.props]: cc/ @jonathanpeppers @Eilon thoughts on this?

File: eng/Version.Details.xml [hunk 5]
```diff
@@ -35,45 +35,37 @@
     <Dependency Name="Microsoft.WindowsAppSDK" Version="0.0.1">
       <Uri>https://dev.azure.com/microsoft/ProjectReunion/_git/ProjectReunionInternal</Uri>
     </Dependency>
-    <Dependency Name="Microsoft.AspNetCore.App.Runtime.win-x64" Version="6.0.3" CoherentParentDependency="Microsoft.Dotnet.Sdk.Internal">
+    <Dependency Name="Microsoft.AspNetCore.Authorization" Version="6.0.4">
```
Comment (TanayParikh) [inline eng/Version.Details.xml:38]: GitHub is choosing to show the diff a bit strangely. So I'll summarize:

Removed:
- Microsoft.AspNetCore.App.Runtime.win-x64
- Microsoft.WindowsDesktop.App.Runtime.win-x64

Also removed the coherant dependencies.

Updated versions to 6.0.4 and updated the hash accordingly.

**Review by jonathanpeppers — APPROVED** [review]
LGTM, maybe CI needs restarted, though

File: eng/Versions.props (hunk 2 above)
Comment (TanayParikh) [reply eng/Versions.props]: Auto-merging now as we need this in ASAP. Basic local testing did not find any compatibility issues. We can revisit as needed.

File: eng/Version.Details.xml (hunk 3 above)
Comment (TanayParikh) [reply eng/Version.Details.xml]: Subscription created & triggered;

```
https://github.com/dotnet/aspnetcore (.NET 6) ==> 'https://github.com/dotnet/maui' ('main')
  - Id: f048ef0c-9fda-42a1-fd39-08da1d875c10
  - Update Frequency: EveryDay
  - Enabled: True
  - Batchable: False
  - PR Failure Notification tags:
  - Merge Policies: []
```

**Pull request merged**